math/src/field/element.rs

```diff
@@ -43,6 +43,18 @@ where
     }
 }
 
+impl<F> FieldElement<F>
```

Comment:
open to alternatives if this is not the right solution? The problem is we get the montgomery representation of the field back from metal and we need to store the `value` as-is when creating back the Field in Rust, instead of converting it to montgomery.

Reply:
It looks good to me.

Maybe in the future we want to just make a big memory copy that sets up N elements, since we will have a huge amount of data and waiting for GPU is slow.